### ISSUE TITLE
Font name ("font-family") with spaces included in name is not supported

### ISSUE THREAD

**Issue by RafaelLinux** [mpdf/mpdf#1715]
### I found this bug, using latest 8.1.2 version
I was mad trying to find why I was trying to use my CSS fonts and mPDF ignored them. As usually, in my five CSS files I declare my elements font this way:
```
font-family: 'Open Sans', sans-serif;
font-family: 'Open Sans Light', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
```
This is usual with fonts in CSS, to include spaces in fonts names.

I tried to use my fonts in mPDF, putting them in a folder and coding:
```
$defaultFontConfig = (new \Mpdf\Config\FontVariables())->getDefaults();
$fontData = $defaultFontConfig['fontdata'];
$mpdf = new \Mpdf\Mpdf(
	[
		'mode' => 'utf-8',
		'format' => 'A6', 
		'aliasNbPg' => '[pagetotal]',    
		'fontDir' => array_merge($fontDirs, ['/var/www/fonts']),
		'fontdata' => $fontData + 
		[
			'Open Sans' => 
			[
				'R' => 'OpenSans-Regular.ttf',
				'L' => 'OpenSans-Light.ttf',
				'C' => 'OpenSans-CondLight.ttf'
			]
    	]
	]
);
```
No errors about, TTF files were located and loaded.

In my HTML body, I coded:
`<body style="font-family: "Open Sans", sans-serif; font-size: 10pt;">';`

However, instead get a PDF with "Open Sans" fonts, I always got embedded "DejaVu" fonts.

After fighting with issue, read a lot of threads in StackOverflow and Google a lot, I guess what's happening: Is an unsupported CSS issue that, however, is not mentioned in "https://mpdf.github.io/css-stylesheets/supported-css.html". Simply, mPDF don't let use standard naming convention to reference fonts names. This means user should use a workaround to avoid change all CSS files using that font.

I just want to ask you to please include this problem in the CSS incompatibilities page to avoid headaches for other users in this regard.

Thank you

**Comment by finwe** [comment]
Isn't this rather the documented quirk of the font name having to be lowercase? See 2. paragraph in example here https://mpdf.github.io/fonts-languages/fonts-in-mpdf-7-x.html? This is planned to be changed in 9.0 https://github.com/mpdf/mpdf/issues/766

**Comment by RafaelLinux** [comment]
Case of font name is another matter. I'm talking about a bunch of fonts includes spaces ("Akzidenz Grotesk", "Gill Sans", "News Roman", "Franklin Gothic", "News Gothic", etc, etc) so CSS programmers use them that way many time ago in their CSS files. mPDF will not recognize that fonts without applying a workaround to avoid change CSS font name in all their sites.
Even in that link, there is not mention to spaces in font names. So I insist, this issue should be included in "Unsupported CSS" page.

**Comment by jakejackson1** [comment]
Spaces or uppercase in the `font-family` declaration in CSS is perfectly fine. However, when registering the font with mPDF you need to use lowercase and no spaces for the key:

```
$defaultFontConfig = (new \Mpdf\Config\FontVariables())->getDefaults();
$fontData = $defaultFontConfig['fontdata'];
$mpdf = new \Mpdf\Mpdf(
	[
		'mode' => 'utf-8',
		'format' => 'A6', 
		'aliasNbPg' => '[pagetotal]',    
		'fontDir' => array_merge($fontDirs, ['/var/www/fonts']),
		'fontdata' => $fontData + 
		[
			'opensans' => 
			[
				'R' => 'OpenSans-Regular.ttf',
				'L' => 'OpenSans-Light.ttf',
				'C' => 'OpenSans-CondLight.ttf'
			]
    	]
	]
);
```

**Comment by finwe** [comment]
From what I've tested, Jake is correct. You can use the full name of the font in font-family declaration as long as the fontdata definition is lowercase letters only.

This works

```
$mpdf = new \Mpdf\Mpdf([
	'fontDir' => array_merge($fontDirs, [__DIR__]),
	'fontdata' => $fontData + 
		[
			'inkfree' => [
				'R' => 'assets/Inkfree.ttf',
			],
		],
]);
```

```
<html>
<head>
	<style>
		.inkfree {
			font-family: "Ink Free";
		}
	</style>
</head>
<body>
	<p style="font-family: \'Ink Free\'">Inkfree line of text</p>

	<p style\'font-family: "Ink Free"\'>Inkfree line of text that is not working</p>

	<p style="font-family: "Ink Free";">Inkfree line of text that is not working</p>

	<p class="inkfree">Inkfree line of text</p>
</body>
</html>
```

Apart from the line using only `"` (as expected) and the line combining `'` and `"` (which is probably another bug).

@RafaelLinux Can you provide a reproducible code example where `font-family: 'Open Sans'` won't work if the fontdata key is defined correctly? If not, we can close this in favor of https://github.com/mpdf/mpdf/issues/766 - I will, in the meantime, emphasise the part of the doc about the font name having to be lowercase letters only.

Given all this, I don't think there is anything to be added to the supported CSS page.

**Comment by RafaelLinux** [comment]
PHP code:
```
$defaultConfig = (new \Mpdf\Config\ConfigVariables())->getDefaults();
$fontDirs = $defaultConfig['fontDir'];

$defaultFontConfig = (new \Mpdf\Config\FontVariables())->getDefaults();
$fontData = $defaultFontConfig['fontdata'];
$mpdf = new \Mpdf\Mpdf(
	[
		'mode' => 'utf-8',
		'format' => 'A4', 
		'aliasNbPg' => '[pagetotal]',    
		'fontDir' => array_merge($fontDirs, ['/var/www/fonts']),

    	'fontdata' => $fontData + 
		[
			'opensans' => ['R' => 'OpenSans-Regular.ttf',],
			'opensanslight' => ['L' => 'OpenSans-Light.ttf',],
			'opensanscondensedlight' => ['C' => 'OpenSans-CondLight.ttf',]
		],
		'default_font' => 'opensans'
	]
);
```

HTML code with in line CSS. Intentionally, I include "serif" as second choice, cause it show clearly in output when a font is not loaded. 

```
<spam style="font-family: \'Open Sans\', serif;">1239 AaGg - Open Sans<br />
<spam style="font-family: \'DejaVu Sans\', serif;">1239 AaGg - DejaVu Sans<br />
<spam style="font-family: \'Open Sans Light\', serif;">1239 AaGg - Open Sans Light<br />
<spam style="font-family: \'Open Sans Condensed Light\', serif;">1239 AaGg - Open Sans Condensed Light<br />
<spam style="font-family: \'DejaVu Sans Condensed\', serif;">1239 AaGg - Deja Vu Condensed Light<br />
```

Generated PDF ALWAYS have only embedded "DejaVu Sans" font, and all text is using that font, despite "Open Sans" fonts are loaded correctly.

**Comment by finwe** [comment]
Your definitions are off. You have to either define all fonts as "regular", or define multiple styles inside a single family and change font variant further in css - by weight. And only possible font variant keys are `R`, `B`, `I` and `BI`, `L` and `C` don't exist.

Your HTML is also off, spam element doesnt exist :) and you are missing closing spam (sic) tags, which can (did not verify) also cause issues.

This works:

```
$defaultConfig = (new \Mpdf\Config\ConfigVariables())->getDefaults();
$fontDirs = $defaultConfig['fontDir'];

$defaultFontConfig = (new \Mpdf\Config\FontVariables())->getDefaults();
$fontData = $defaultFontConfig['fontdata'];
$mpdf = new \Mpdf\Mpdf(
	[
		'mode' => 'utf-8',
		'format' => 'A4', 
		'aliasNbPg' => '[pagetotal]',    
		'fontDir' => array_merge($fontDirs, [__DIR__ . '/fonts']),
		'fontdata' => $fontData + 
		[
			'opensans' => ['R' => 'OpenSans-Regular.ttf',],
			'opensanslight' => ['R' => 'OpenSans-Light.ttf',],
			'opensanscondensedlight' => ['R' => 'OpenSans-CondensedLight.ttf',]
		],
		'default_font' => 'opensans'
	]
);

$html = '

<span style="font-family: \'Open Sans\', serif;">1239 AaGg - Open Sans</span><br />
<span style="font-family: \'DejaVu Sans\', serif;">1239 AaGg - DejaVu Sans</span><br />
<span style="font-family: \'Open Sans Light\', serif;">1239 AaGg - Open Sans Light</span><br />
<span style="font-family: \'Open Sans Condensed Light\', serif;">1239 AaGg - Open Sans Condensed Light</span><br />
<span style="font-family: \'DejaVu Sans Condensed\', serif;">1239 AaGg - Deja Vu Condensed Light</span><br />

';

$mpdf->WriteHTML($html);
$mpdf->Output('', 'I');
```

**Comment by RafaelLinux** [comment]
You have reason about my HTML syntax. I tried to do example ASAP and had errors. Same about "Styles" definition ... I presumed "C" could be used for "Condensed". My fault again.
If you are reading me here, is because however, despite your clear example, for me thing doesn't work same way.

I must first say that I checked again that my fonts are being loaded. Apache shows an error message when trying to USE a font not found:

`Uncaught Mpdf\\MpdfException: Cannot find TTF TrueType font file "OpenSans-egular.ttf" in configured font directories.`

and I have seen that mPDF seems to check fonts added only when trying to use it. I know that when I notices that one of my fonts filename is "OpenSans-CondLight.ttf" not "OpenSans-CondensedLight.ttf" and Apache gave not erros at all (cause I was not trying to use that font).

Using your fixed code, I got text using DejaVu correctly, but text that should be using "Open Sans" is printed as serif. I don't understand why "Open Sans" font even are not embedded in the PDF and if there is a way to force it, but could be issue, I don't know what more to try.

Thank you anyway for trying to help.
